application-analytics-default/src/main/java/com/xwiki/analytics/script/AnalyticsScriptService.java

```diff
@@ -71,6 +85,40 @@ public JsonNode makeRequest(Map<String, String> parameters, Map<String, String>
         }
     }
 
+    /**
+     * Endpoint for all the aggregator jobs.
+     *
+     * @param hint for which aggregator you want to use
+     */
+    public void aggregate(String hint)
+    {
+
+        for (Aggregator aggregator : aggregators.get()) {
+            if (hint.equals(aggregator.getHint())) {
+                aggregator.aggregateData();
+                return;
+            }
+        }
+    }
+
+    /**
+     * Endpoint for handling aggregated data for livedata.
+     * @param dataSource page where the data is stored
+     * @param asc if you want the data to be in ascending or descending order
+     * @param sortField the field that you want to sort after
+     * @param filters map where the keys are the fields and the values are the text that you want to filter after
+     * @param pageSize how many elements are on a page
+     * @param pageCount current page offset
+     * @return a list with jsons
+     * @throws JsonProcessingException thrown if the dataSource doesn't store the data in JSON format
+     * @throws XWikiException
+     */
+    public Pair<Integer, List<JsonNode>> handleData(DocumentReference dataSource, String asc, String sortField,
```

Comment:
"handleData" sounds very generic. Maybe something like "retrieveAggregatedData" would be more descriptive - or something similar.

Reply:
I think you should indentify the data store using the same method that you use to aggregate the data. You start the aggregation job by using a macro identifier but you retrieve the said data by using a document reference. I think that both should be identified using the same method. Thus, the data store should either be configurable by the developer than implements the aggregator or (i.e. the aggregator interface should offer a "getStoreLocation")  there should be a generic store for each aggregator identified by the id (i.e. the data aggregated by component `component_name` should be located at `MatomoIntegration/AggregatedData/component_name`).